### PR TITLE
generate correct serials on Python 3

### PR DESCRIPTION
**Pull request by evgeni**
fixSerial is supposed to make the serial have an even number of digits.

On Python2 dividing by an integer yields a different result than a
float:

    >>> 5/2.0
    2.5
    >>> 5/2
    2

But the same does not hold true for Python3:

    >>> 5/2.0
    2.5
    >>> 5/2
    2.5

Let's use modulo (%) to determine if the number is even.

### CODE REVIEW

**Comment by evgeni** [comment]
I am not sure why the tests didn't catch that. They properly fail locally for me:

```
+ katello-ssl-tool --gen-client -p file:/etc/pki/katello/private/katello-default-ca.pwd --ca-cert-dir /etc/pki/katello-certs-tools/certs --set-common-name foo.example.com --ca-cert katello-default-ca.crt --ca-key katello-default-ca.key --set-country US --set-state 'North Carolina' --set-city Raleigh --set-org FOREMAN --set-org-unit FOREMAN_PROXY --cert-expiration 36500 --server-rpm be01a30e2769-foreman-proxy-client

Generating the web server's SSL private key: ./ssl-build/be01a30e2769/server.key

Generating web server's SSL certificate request: ./ssl-build/be01a30e2769/server.csr
Using distinguished names:
    --set-country      = "US"
    --set-state        = "North Carolina"
    --set-city         = "Raleigh"
    --set-org          = "FOREMAN"
    --set-org-unit     = "FOREMAN_PROXY"
    --set-hostname     = "be01a30e2769"
    --set-email        = "admin@example.com"
Rotated: katello-ca-openssl.cnf --> katello-ca-openssl.cnf.1

Generating/signing web server's SSL certificate: server.crt

ERROR: web server's SSL certificate generation/signing failed:

/bin/sh: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
Using configuration from /tmp/tmp.n6N6EchWAo/ssl-build/katello-ca-openssl.cnf
unable to load number from /tmp/tmp.n6N6EchWAo/ssl-build/serial
error while loading serial number
140349333038912:error:0D066091:asn1 encoding routines:a2i_ASN1_INTEGER:odd number of chars:crypto/asn1/f_int.c:103:
```

**Comment by evgeni** [comment]
Oooh, they sometimes fail, and sometimes don't. Depending on what serial was generated. Makes sense :vomiting_face:

**Comment by evgeni** [comment]
There, a reliably broken test. Yay.

File: tests/bad-serial.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+DIRECTORY=$(mktemp -d)
+trap "rm -rf $DIRECTORY" EXIT
+cd $DIRECTORY
+
+set -ex
+
+mkdir -p ssl-build/
+# generate a definetly bad serial
```
Comment (ekohl) [inline tests/bad-serial.sh]:
```suggestion
# generate a definitely bad serial
```

File: tests/bad-serial.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# we need to generate one that will be higher than the one the CA uses (and that's autogenetated)
```
Comment (ekohl) [inline tests/bad-serial.sh]:
```suggestion
# we need to generate one that will be higher than the one the CA uses (and that's autogenerated)
```

**Review by ekohl — APPROVED** [review]
I've really been wondering about the whole fixSerial code and have tried to full remove it, but I guess this at least makes it less bad.

**Pull request merged**